confignode/src/main/java/org/apache/iotdb/confignode/client/async/datanode/AsyncDataNodeClientPool.java

```diff
@@ -183,6 +224,31 @@ public void sendAsyncRequestToDataNodeWithRetry(
     }
   }
 
+  public void sendAsyncRequestToDataNodeWithRetry(Object req, AbstractRetryHandler handler) {
+    Map<Integer, TDataNodeLocation> dataNodeLocationMap = handler.getDataNodeLocationMap();
+    if (dataNodeLocationMap.isEmpty()) {
+      return;
+    }
+    for (int retry = 0; retry < MAX_RETRY_NUM; retry++) {
+      CountDownLatch countDownLatch = new CountDownLatch(dataNodeLocationMap.size());
+      handler.setCountDownLatch(countDownLatch);
+      for (TDataNodeLocation targetDataNode : dataNodeLocationMap.values()) {
+        handler.setTargetDataNode(targetDataNode);
+        sendAsyncRequestToDataNode(targetDataNode, req, handler, retry);
+      }
+
+      try {
+        countDownLatch.await();
+      } catch (InterruptedException e) {
+        LOGGER.error("Interrupted during {} on ConfigNode", handler.getDataNodeRequestType());
+      }
+      // Check if there is a node that fails to send the request, and retry if there is one
+      if (dataNodeLocationMap.isEmpty()) {
+        break;
+      }
+    }
```

Comment:
Normally, we construct a new handler for every asynchronize request instead of using the same one. Otherwise, you should add concurrency control in your handler~

Reply:
Fixed. PTAL